### PR TITLE
Eliminate deprecation warnings in tests

### PR DESCRIPTION
**Pull request by jdinan**

### CODE REVIEW

**Comment by jdinan** [comment]
Note: I added an "ENABLE_DEPRECATED_TESTS" preprocessor variable, but didn't actually hook it up to the build system.  It would have to flow in through CFLAGS/CXXFLAGS since we don't include the configury-generated header in the tests..

**Comment by jdinan** [comment]
New warnings popping up in Travis, related to the C++ generics:
```
In file included from ../../../test/unit/cxx_test_shmem_get.cpp:37:0:
../../mpp/shmem.h:2566:91: warning: ‘deprecated’ attribute directive ignored [-Wattributes]
 SHMEM_CXX_ATTRIBUTE_DEPRECATED static inline void shmem_add(int *target, int value, int pe) {
                                                                                           ^
../../mpp/shmem.h:2569:93: warning: ‘deprecated’ attribute directive ignored [-Wattributes]
 SHMEM_CXX_ATTRIBUTE_DEPRECATED static inline void shmem_add(long *target, long value, int pe) {
                                                                                             ^
```

**Comment by jdinan** [comment]
That should fix it.  Problem was that ```[[deprecated]]``` was introduced in C++14, but its use was gated by a check for C++11.

**Comment by jdinan** [comment]
Yup -- that warning message is gone now.  Wow, "introducted" (in cacaf4c commit message), I am crushing vocab today.

**Comment by jdinan** [comment]
@davidozog This is ready.  Sorry, found a few more warnings that I wanted to squash while I was reviewing the Travis logs.

**Comment by jdinan** [comment]
Closes #471

File: mpp/shmem.h4.in [hunk 1]
```diff
@@ -99,6 +99,12 @@ include(shmem_c_func.h4)dnl
 #ifdef __cplusplus
 } /* extern "C" */
 
+#if __cplusplus >= 201402L
+#define SHMEM_CXX_ATTRIBUTE_DEPRECATED [[deprecated]]
+#else
+#define SHMEM_CXX_ATTRIBUTE_DEPRECATED
```
Comment (davidozog) [inline mpp/shmem.h4.in]: Doesn't `__attribute__ ((deprecated))` work for pre-C++14?  If so, might be able to do:
`#define SHMEM_CXX_ATTRIBUTE_DEPRECATED SHMEM_C_ATTRIBUTE_DEPRECTATED`?

File: test/unit/c11_test_shmem_atomic_add.c [hunk 2]
```diff
@@ -41,6 +41,14 @@
 enum op { ADD = 0, ATOMIC_ADD, CTX_ATOMIC_ADD, FADD, ATOMIC_FETCH_ADD,
           CTX_ATOMIC_FETCH_ADD };
 
+#ifdef ENABLE_DEPRECATED_TESTS
```
Comment (davidozog) [inline test/unit/c11_test_shmem_atomic_add.c]: Might want to file an issue to hook this in so we don't forget?  (or not!  :hear_no_evil:)

File: mpp/shmem.h4.in (hunk 1 above)
Comment (jdinan) [reply mpp/shmem.h4.in]: Kinda, but not really..  🙁  I tried, but couldn't find a portable location for the attribute on a function definition (versus declaration).

File: test/unit/c11_test_shmem_atomic_add.c (hunk 2 above)
Comment (jdinan) [reply test/unit/c11_test_shmem_atomic_add.c]: Done: #542

**Comment by davidozog** [comment]
Ready to merge if you are, @jdinan

**Pull request merged**